provisioning_client/deps/CMakeLists.txt

```
@@ -14,41 +14,60 @@ else()
+            ./RIoT/Simulation/RIoT/Core/RIoTCrypt/ecdh.c
+            ./RIoT/Simulation/RIoT/Core/RIoTCrypt/ecdsa.c
+            ./RIoT/Simulation/RIoT/Core/RIoTCrypt/ecp.c
+            ./RIoT/Simulation/RIoT/Core/RIoTCrypt/ecp_curves.c
+            ./RIoT/Simulation/RIoT/Core/RIoTCrypt/hkdf.c
+            ./RIoT/Simulation/RIoT/Core/RIoTCrypt/hmac_drbg.c
+            ./RIoT/Simulation/RIoT/Core/RIoTCrypt/md.c
+			./RIoT/Simulation/RIoT/Core/RIoTCrypt/platform_util.c
+            ./RIoT/Simulation/RIoT/Core/RIoTCrypt/sha1.c
+            ./RIoT/Simulation/RIoT/Core/RIoTCrypt/sha256.c
+            ./RIoT/Simulation/RIoT/Core/RIoTCrypt/x509bldr.c
         )
 
         set(msr_riot_h_files
-            ./RIoT/Emulator/DICE/DiceSha256.h
-            ./RIoT/Emulator/RIoT/RIoT.h
-            ./RIoT/Emulator/RIoT/RiotStatus.h
-            ./RIoT/Emulator/RIoT/RIoTCrypt/include/RiotAes128.h
-            ./RIoT/Emulator/RIoT/RIoTCrypt/include/RiotCrypt.h
-            ./RIoT/Emulator/RIoT/RIoTCrypt/include/RiotDerEnc.h
-            ./RIoT/Emulator/RIoT/RIoTCrypt/include/RiotEcc.h
-            ./RIoT/Emulator/RIoT/RIoTCrypt/include/RiotHmac.h
-            ./RIoT/Emulator/RIoT/RIoTCrypt/include/RiotKdf.h
-            ./RIoT/Emulator/RIoT/RIoTCrypt/include/RiotSha256.h
-            ./RIoT/Emulator/RIoT/RIoTCrypt/include/RiotTarget.h
-            ./RIoT/Emulator/RIoT/RIoTCrypt/include/RiotX509Bldr.h
-            ./RIoT/Emulator/RIoT/RIoTCrypt/include/RiotBase64.h
+            ./RIoT/Simulation/DICE/DiceSha256.h
+            ./RIoT/Simulation/RIoT/Core/RIoT.h
+            ./RIoT/Simulation/RIoT/Core/RiotStatus.h
+            ./RIoT/Simulation/RIoT/Core/RIoTCrypt/include/RiotCrypt.h
+            ./RIoT/Simulation/RIoT/Core/RIoTCrypt/include/base64.h
+            ./RIoT/Simulation/RIoT/Core/RIoTCrypt/include/derenc.h
+			./RIoT/Simulation/RIoT/Core/RIoTCrypt/include/x509bldr.h
+            ./RIoT/Simulation/RIoT/Core/RIoTCrypt/include/mbedtls/aes.h
+            ./RIoT/Simulation/RIoT/Core/RIoTCrypt/include/mbedtls/ecdh.h
+            ./RIoT/Simulation/RIoT/Core/RIoTCrypt/include/mbedtls/ecdsa.h
+            ./RIoT/Simulation/RIoT/Core/RIoTCrypt/include/mbedtls/ecp.h
+			./RIoT/Simulation/RIoT/Core/RIoTCrypt/include/mbedtls/bignum.h
+			./RIoT/Simulation/RIoT/Core/RIoTCrypt/include/mbedtls/hkdf.h
+            ./RIoT/Simulation/RIoT/Core/RIoTCrypt/include/mbedtls/hmac_drbg.h
+			./RIoT/Simulation/RIoT/Core/RIoTCrypt/include/mbedtls/md.h
+			./RIoT/Simulation/RIoT/Core/RIoTCrypt/include/mbedtls/platform.h
+			./RIoT/Simulation/RIoT/Core/RIoTCrypt/include/mbedtls/platform_util.h
+            ./RIoT/Simulation/RIoT/Core/RIoTCrypt/include/mbedtls/sha1.h
+            ./RIoT/Simulation/RIoT/Core/RIoTCrypt/include/mbedtls/sha256.h
         )
 
-        include_directories(./RIoT/Emulator/DICE)
-        include_directories(./RIoT/Emulator/RIoT)
-        include_directories(./RIoT/Emulator/RIoT/RIoTCrypt/include)
+        include_directories(./RIoT/Simulation/DICE)
```

These should probably be `target_include_directories()` instead. Everything should be `target_` based if possible. 
```suggestion
        target_include_directories(msr_riot PUBLIC
            ${CMAKE_CURRENT_LIST_DIR}/RIoT/Simulation/DICE
        )
```

---

The same goes for any `include_directories` that you have added. So for 58-61 it would be 
```
target_include_directories(msr_riot
    PUBLIC
        ${CMAKE_CURRENT_LIST_DIR}/RIoT/Simulation/DICE
        ${CMAKE_CURRENT_LIST_DIR}/RIoT/Simulation/RIoT/Core
        ${CMAKE_CURRENT_LIST_DIR}/RIoT/Simulation/RIoT/Core/RIoTCrypt/include
        ${CMAKE_CURRENT_LIST_DIR}/RIoT/Simulation/RIoT/Core/RIoTCrypt/include/mbedtls
)
```

Just as an explanation for why...
Non target versions of CMake commands apply to anything at or below the current CMakeLists.txt scope. So that goes for compile_definitions, include_directories, etc. The `target_` versions should ALWAYS be preferred over non because it keeps the scope of compilation requirements as tight as possible and doesn't corrupt the compilation of other libs/targets. There may be exceptions to the rule but there has to be good reason to.

---

Fixed, thanks for the explanation will file away for future use